internal/providers/terraform/google/sql_database_instance.go

```diff
@@ -1,305 +1,38 @@
-			VendorName: strPtr("gcp"),
-			Region:     strPtr(region),
-			Service:    strPtr("Cloud SQL"),
-			AttributeFilters: []*schema.AttributeFilter{
-				{Key: "resourceGroup", Value: strPtr(resourceGroup)},
-				{Key: "description", ValueRegex: strPtr(descriptionRegex)},
-			},
-		},
-	}
-}
-
-func sqlInstanceDBVersionToDBType(dbVersion string) SQLInstanceDBType {
-	if strings.Contains(dbVersion, "POSTGRES") {
-		return PostgreSQL
-	} else if strings.Contains(dbVersion, "MYSQL") {
-		return MySQL
-	} else if strings.Contains(dbVersion, "SQLSERVER") {
-		return SQLServer
+func NewSQLDatabaseInstance(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	r := &google.SQLDatabaseInstance{Address: d.Address, DatabaseVersion: d.Get("database_version").String(), Tier: d.Get("settings.0.tier").String(), Region: d.Get("region").String()}
+	if !d.IsEmpty("settings.0.ip_configuration.0.ipv4_enabled") {
+		r.UseIPV4 = d.Get("settings.0.ip_configuration.0.ipv4_enabled").Bool()
 	} else {
-		return MySQL
-	}
-}
-
-func sqlInstanceTierToResourceGroup(tier string) string {
-	data := map[string]string{
-		"db-f1-micro": "SQLGen2InstancesF1Micro",
-		"db-g1-small": "SQLGen2InstancesG1Small",
+		r.UseIPV4 = true // Should use ipv4 if the ipv4_enabled is empty
 	}
-
-	if data[tier] != "" {
-		return data[tier]
-	} else if strings.Contains(tier, "db-n1-standard") {
-		return "SQLGen2InstancesN1Standard"
-	} else if strings.Contains(tier, "db-n1-highmem") {
-		return "SQLGen2InstancesN1Highmem"
-	}
-
-	return ""
-}
-
-func sqlInstanceTypeToDescriptionName(dbType SQLInstanceDBType) string {
-	dbTypeNames := map[SQLInstanceDBType]string{
-		MySQL:      "MySQL",
-		PostgreSQL: "PostgreSQL",
-		SQLServer:  "SQL Server",
-	}
-
-	return dbTypeNames[dbType]
-}
-
-func availabilityTypeDescName(availabilityType string) string {
-	availabilityTypeNames := map[string]string{
-		"REGIONAL": "Regional",
-		"ZONAL":    "Zonal",
-	}
-
-	return availabilityTypeNames[availabilityType]
-}
-
-func sqlInstanceAvDBTypeToDescription(availabilityType string, dbType SQLInstanceDBType) string {
-	dbTypeString := sqlInstanceTypeToDescriptionName(dbType)
-	availabilityTypeString := availabilityTypeDescName(availabilityType)
-
-	description := fmt.Sprintf("%s: %s", dbTypeString, availabilityTypeString)
-
-	return description
-}
-
-func sqlInstanceStorage(region string, dbType SQLInstanceDBType, availabilityType, diskType string, diskSizeGB int64) *schema.CostComponent {
-	diskTypeHumanReadableNames := map[string]string{
-		"PD_SSD": "SSD",
-		"PD_HDD": "HDD",
+	if !d.IsEmpty("replica_configuration") {
+		r.ReplicaConfiguration = d.Get("replica_configuration").String()
 	}
-
-	diskTypeAPIResourceGroup := map[string]string{
-		"PD_SSD": "SSD",
-		"PD_HDD": "PDStandard",
+	if !d.IsEmpty("settings.0.availability_type") {
+		r.AvailabilityType = d.Get("settings.0.availability_type").String()
 	}
-
-	if dbType == SQLServer {
-		diskType = "PD_SSD"
+	if !d.IsEmpty("settings.0.disk_type") {
```

Comment:
As above, I don't think this empty check is needed.

Reply:
Applied